packages/relay/src/lib/precheck.ts

```diff
@@ -152,6 +158,15 @@ export class Precheck {
     }
   }
 
+  /**
+   * Checks if a transaction is the Foundry deterministic deployment transaction.
+   * @param {Transaction} tx - The transaction to check.
+   * @returns {boolean} Returns true if the transaction is the Foundry deterministic deployment transaction, otherwise false.
+   */
+  static isFoundryDeterministicDeploymentTransaction(tx: Transaction): boolean {
```

Comment:
Add coverage in the precheck.spec.ts
Always add unit tests first. Then component/integration level tests, then acceptance/E2E tests

Reply:
Copy!